include/tvm/internal/VariableCountingVector.h

```diff
@@ -0,0 +1,76 @@
+/** Copyright 2017-2020 CNRS-AIST JRL and CNRS-UM LIRMM */
+
+#pragma once
+
+#include <tvm/api.h>
+#include <tvm/defs.h>
+
+#include <tvm/VariableVector.h>
+#include <tvm/internal/RangeCounting.h>
+#include <tvm/utils/internal/map.h>
+
+namespace tvm::internal
+{
+/** This class adds a counting logic over a VariableVector, that allows to add
+ * and remove subvariables without constraints, and tracks exactly how many
+ * time a (part of) variable was added and removed.
+ * This is in particular useful for computing the variables of a problem, where
+ * different part of a same variables can be added by different functions.
+ */
```

Comment:
I think the documentation should add a note about what's the motivation/usage for this class in the framework

Reply:
Done in 1bea6bb15bcd45398656e2ad06cc6edf68575c82